### ISSUE TITLE
Add support for ES2015 'new.target'

### ISSUE THREAD

**Issue by fatcerberus** [svaarala/duktape#1014]
See https://github.com/svaarala/duktape/pull/1013#issuecomment-253958740

Just posting this here as a placeholder so I don't forget about it, as I'd like to look into implementing support for this at some point.

**Comment by fatcerberus** [comment]
According to the specification `new.target` is looked up from the containing lexical environment:
http://www.ecma-international.org/ecma-262/6.0/#sec-getnewtarget

Best I can tell, scoping semantics are the same as for `this`.  Duktape doesn't yet support arrow functions, so in practice that means "the containing function".

**Comment by svaarala** [comment]
That's right, it should be similar to `this` (just requires parsing a bit differently) unless one plans ahead for arrow functions.

For the case of bound functions, I'm not sure if `new.target` should be the bound function or the ultimate non-bound target. Activations currently track the non-bound function but not the original bound function, which is not available during execution now.

**Comment by fatcerberus** [comment]
Quick test in Chrome:

```js
> function O() { return new.target; }
undefined
> new O()
O() { return new.target; }
> var P = O.bind(window);
undefined
> P === O
false
> new P() === O
true
> new P() === P
false
```

If Chrome is correct then for bound functions it appears that `new.target` refers to the non-bound target at the end of the chain.

**Comment by svaarala** [comment]
That would be very nice to avoid extra effort :)

**Comment by svaarala** [comment]
This was implemented in 2.2.0 minimally with some limitations, leftovers are tracked in #1547.
